inference-api/python-client/client.py

```diff
@@ -0,0 +1,91 @@
+import requests
+
+from asgiref.sync import async_to_sync
+from retry import retry
+from typing import Optional
+
+from .exceptions import InvalidInferenceAPIClientArgsException
+from .types import Inference
+from .django_jwt_client import DjangoJWTClient
+
+class InferenceAPIClient:
+  AUTH_PATH = '/api/auth/jwt_login'
+  GET_INFERENCE_PATH = 'models/{model_id}/inferences/{study_instance_uid}/{series_instance_uid}'
+
+  def __init__(
+    self,
+    private_key_filename: Optional[str] = None,
+    private_key_string: Optional[str] = None,
+    base_url: str = 'https://api.bunkerhillhealth.com/',
+  ) -> None:
+    self._validate_args(private_key_filename, private_key_string)
+    if private_key_filename:
+      with open(private_key_filename, 'r') as f:
+        client_private_key = f.read()
+    elif private_key_string:
+      client_private_key = private_key_string
+    self._django_jwt_client = DjangoJWTClient(
+      client_private_key=client_private_key,
+      base_url=base_url,
+      auth_path=self.AUTH_PATH,
+    )
+
+  def _validate_args(
+    self,
+    private_key_filename: Optional[str] = None,
+    private_key_string: Optional[str] = None,
+  ) -> None:
+    if private_key_filename is None and private_key_string is None:
+      raise InvalidInferenceAPIClientArgsException(
+        reason='either private_key_filename or private_key_string must be provided')
+    if private_key_filename is not None and private_key_string is not None:
+      raise InvalidInferenceAPIClientArgsException(
+        reason='private_key_filename and private_key_string cannot both be provided')
+
+  async def get_inference_async(
+    self,
+    model_id: str,
+    study_instance_uid: str,
+    series_instance_uid: str,
+    segmentation_destination_filename: str,
+  ) -> Inference:
+    resource_path = self.GET_INFERENCE_PATH.format(
+      model_id=model_id,
+      study_instance_uid=study_instance_uid,
+      series_instance_uid=series_instance_uid,
+    )
+    response_json = await self._django_jwt_client.get_json(resource_path)
+    self._download_segmentation(
+      presigned_url=response_json['segmentation_presigned_url'],
+      destination_filename=segmentation_destination_filename,
+    )
```

Comment:
this should be in a try catch so we can return an error code if we fail on either getting the data from the presigned URL or have an issue with writing the file (destination does not exist, don't have proper access, etc.)